### ISSUE TITLE
IDISP015 should be reported on the problematic returned expressions

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/IDisposableAnalyzers#104]
Version: 2.0.3.1

Even though the `return NotInstalled;` statement is causing the IDISP015 "Member should not return created and cached instance.", the diagnostic is reported on the name `Foo`, even though I'd expect it to be reported on the expressions `new DisposableObject()` and `NotInstalled` inside the return statements:

```cs
    private static readonly IDisposable NotInstalled = new NoOpObject();

    public static IDisposable Foo()
    {
        if (IsInstalled)
        {
            return new DisposableObject();
        }

        return NotInstalled;
    }
```

The downside of using the method identifier is that any pragma suppression (and explanation) is not lexically near the cause of the diagnostic.
